staging/src/k8s.io/apiserver/pkg/storage/cacher/cacher.go

```diff
@@ -532,9 +532,18 @@ func (c *Cacher) Watch(ctx context.Context, key string, opts storage.ListOptions
 		return nil, err
 	}
 
-	readyGeneration, err := c.ready.waitAndReadGeneration(ctx)
-	if err != nil {
-		return nil, errors.NewServiceUnavailable(err.Error())
+	var readyGeneration int
+	if utilfeature.DefaultFeatureGate.Enabled(features.ResilientWatchCacheInitialization) {
+		var ok bool
+		readyGeneration, ok = c.ready.checkAndReadGeneration()
+		if !ok {
+			return nil, errors.NewTooManyRequests("storage is (re)initializing", 1)
```

Comment:
why not add some jitter so that clients won't come back at roughly the same time?

Reply:
Our headers only support int values for now. So we can't really jitter it.
Supporting fractional values was always desired but was never prioritized.